standard/variables.md

```diff
@@ -988,3 +998,185 @@ variable_reference
+
+> *Example:* The following example demonstrates a local reference variable whose referent is an element of an array:
+>
+> ```csharp
+> public class C
+> {
+>
+>     public void M()
+>     {
+>         int[] arr = new int[10];
+>         // element is a reference variable that refers to arr[5]
+>         ref int element = ref arr[5];
+>         element += 5; // arr[5] has been incremented by 5
+>     }     
+> }
+> ```
+>
+> *end example*
+
+A ***reference return*** is the expression returned by reference from a method whose return type includes the `ref` or `ref readonly` modifiers (§15.6.1). The variable expression of the reference return is the referent of the reference return.
+
+> *Example:* The following example demonstrates a reference return whose referent is an element of an array field:
+>
+> ```csharp
+> public class C
+> {
+>     private int[] arr = new int[10];
+>
+>     public readonly ref int M()
+>     {
+>         // element is a reference variable that refers to arr[5]
+>         ref int element = ref arr[5];
+>         return ref element; // return reference to arr[5];
+>     }     
+> }
+> ```
+>
+> *end example*
+
+### §ref-safe-contexts Ref safe contexts
+
+All reference variables obey safety rules that ensure the context of the reference variable is not greater than the ref-safe-context of its referent.
+
+For any variable, the ***ref-safe-context*** of that variable is the context where a *variable_reference* (§9.5) to that variable is valid. The referent of a reference variable must have a ref-safe-context that is at least as wide as the context of the reference variable.
+
+> *Note*: The compiler determines the safe context through a static analysis of the program text. The safe context reflects the lifetime of a variable at runtime. *end note*
+
+There are three valid ref-safe-contexts:
+
+- ***declaration-block***: A *variable_reference* to a local variable declared in a block is valid from its declaration to the end of the block in which it is declared. Each nested block represents a different context.  The ref-safe-context of a local variable is the declaration-block in which it is declared. A local variable declared in a method has a ref-safe-context of the block that defines the method. A variable declared in a block is a valid referent only if the reference variable is declared in the same block after the referent, or a nested block.
+- ***function-member***: A *variable_reference* to a value parameter on a function member declaration, including the implicit `this` parameter, is valid in the entire function member. The ref-safe-context of the fields of a `struct` type is function-member. A variable with ref-safe-context of function-member is a valid referent only if the reference variable is declared in the same function member.
```

Comment:
The reason is because the `this` parameter for a `struct` method is passed-by-value. So, `this` is *a copy* of the `struct` passed as an argument.

Unless what you mean is to remove "entire" from the previous sentence?

Reply:
> The reason is because the `this` parameter for a `struct` method is passed-by-value.

The `this` parameter in a struct instance method is a `ref`. Did I misunderstand what this sentence was referring to?

```cs
var s = new S { F = 0 };
s.M();
Console.WriteLine(s.F); // 1

struct S
{
    public int F;
    public void M() { F = 1; }
}
```

Reply:
It is a ref parameter, but a special one with special rules in this version. It is not returnable. In C# 11 the `this` parameter in struct methods is equivalent to a `scoped ref` parameter.